sot_hpp/supervisor.py

```diff
@@ -120,7 +124,7 @@ def makeInitialSot (self):
         sot.setSize(self.sotrobot.dynamic.getDimension())
         self.keep_posture = Posture ("posture_keep", self.sotrobot)
         self.keep_posture.tp.setWithDerivative (False)
-        self.keep_posture._signalPositionRef().value = self.sotrobot.dynamic.position.value
+        self.keep_posture._signalPositionRef().value = tuple([-0.74, 0.0, 1.0, 0.0, 0.0, 0.0] + list(self.sotrobot.dynamic.position.value)[6:])
```

Comment:
Why do you need this values ?

Reply:
The COM of the robot in the HPP frame is at those coordinates (approx.).
But the keep_posture task is « internally » (there is no actuator able to directly move the COM, but the controller in the task is computing controls anyway, and integrating them) moving the computed COM to its reference value which is (0, 0, 0).
So, when we send the goal coordinates of the feet from HPP to the SoT, there is an offset of 0,74m between the goal  and the current position of the feet. This was the cause of the strange feet movements at the beginning of the demo.

I do agree that this is a dirty hack. Maybe we can get the COM position and orientation from HPP at the beginning of the trajectory to initialize `self.sotrobot.dynamic.position.value`

Reply:
This is a rather annoying problem. Could you add a comment with a TODO mentioning your explanation ?